finetune/full.py

```diff
@@ -97,31 +98,40 @@ def main(fabric: L.Fabric, data_dir: Path, checkpoint_dir: Path, out_dir: Path)
     model = fabric.setup_module(model)
     optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate, weight_decay=weight_decay)
     optimizer = fabric.setup_optimizers(optimizer)
-
-    load_checkpoint(fabric, model, checkpoint_path)
+    state = {"model": model, "optimizer": optimizer, "hparams": hparams, "iter_num": 0, "step_count": 0,
+             "total_lengths": 0}
+
+    if resume is True:
+        resume = sorted(out_dir.glob("*ckpt.pth"))[-1]
```

Comment:
Let's update this line with this: https://github.com/Lightning-AI/lit-gpt/blob/ad6554911951768b1db275a9b911e767faaa9aa7/pretrain/tinyllama.py#L113

So we handle the sorting of numberings in the name correctly.